src/connmgr.cpp

```diff
@@ -62,6 +62,22 @@ NodeId CConnMgr::NextNodeId()
     return ++next;
 }
 
+/**
+ * Given a node ID, return a node reference to the node.
+ */
+CNodeRef CConnMgr::FindNodeFromId(NodeId id)
+{
+    LOCK(cs_vNodes);
+
+    BOOST_FOREACH (CNode *pNode, vNodes)
+    {
```

Comment:
Could do, though I've been cautious because of backports.  But connmgr is dev only so here it would be OK.